src/main/java/org/broadinstitute/consent/http/resources/DacResource.java

```diff
@@ -206,6 +212,37 @@ public Response filterUsers(@PathParam("term") String term) {
         return Response.ok().entity(users).build();
     }
 
+    @PUT
+    @Consumes("application/json")
+    @Produces("application/json")
+    @Path("{dacId}/dataset/{datasetId}")
+    @RolesAllowed({CHAIRPERSON})
+    public Response approveDataset(@Auth AuthUser authUser, @PathParam("dacId") Integer dacId, @PathParam("datasetId") Integer datasetId, String json) {
+        try{
+            User user = userService.findUserByEmail(authUser.getEmail());
+            Dataset dataset = datasetService.findDatasetById(datasetId);
+            if(dataset.getDacId() != dacId) { 
```

Comment:
```suggestion
            if(Objects.isNull(dataset) || !Objects.equals(dataset.getDacId(), dacId)) { 
```

Response message is different when the dataset actually doesn't exist since returned dataset is null and  `dataset.getDacId()` throws an error. 

![Screenshot (102)](https://user-images.githubusercontent.com/81024249/187503067-e6d4ef45-a964-450b-b0d8-0e288fbdd8ff.png)